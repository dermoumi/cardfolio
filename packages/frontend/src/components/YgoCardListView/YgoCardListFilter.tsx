import type { YgoCardFilters } from "@/types";
import type { FC } from "react";

import { useCallback } from "react";

import TextInput from "../TextInput";
import styles from "./YgoCardListFilter.module.css";

export type YgoCardListFilterProps = {
  filters: YgoCardFilters;
  setFilters: (filter: YgoCardFilters) => void;
};

const YgoCardListFilter: FC<YgoCardListFilterProps> = ({ filters, setFilters }) => {
  const setNameFilter = useCallback((name: string) => {
    setFilters({ ...filters, name });
  }, [setFilters, filters]);

  return (
    <div className={styles.filterBar}>
      <TextInput
        type="search"
        placeholder="Search cards..."
        value={filters.name || ""}
        onChange={setNameFilter}
      />
    </div>
  );
};

export default YgoCardListFilter;
