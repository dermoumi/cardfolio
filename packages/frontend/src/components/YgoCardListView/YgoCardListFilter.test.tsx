import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import YgoCardListFilter from "./YgoCardListFilter";

describe("YgoCardListFilter", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(
      <YgoCardListFilter filters={{ name: "test" }} setFilters={vi.fn()} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
