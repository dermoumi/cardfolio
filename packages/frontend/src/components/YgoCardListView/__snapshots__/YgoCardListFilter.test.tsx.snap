// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`YgoCardListFilter > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="_filterBar_4db3c9"
  >
    <input
      name="card-name"
      placeholder="Search cards..."
      type="search"
      value="test"
    />
  </div>
</DocumentFragment>
`;
